### DISCUSSION TITLE
Workflow dispatch API doesn't return the run id

### DISCUSSION THREAD

**Discussion by benc-uk** [community/community#126578] [Apps, API and Webhooks] [answered]
### Select Topic Area

Product Feedback

### Body

I'm the developer and maintainer of a Action that uses the workflow-dispatch event and API to allow people to trigger other workflows. I've had several requests for features that require the details of the run that was triggered. However the API doesn't return this information, in fact it returns nothing!

https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event

This seems a really strange decision and there's no other means to query the runs to get the one you triggered (without the risk you find the wrong one, which was one tiggered by someone else or some other event)

Can we have this API return the run id? Or provide some other means to get the run id

**Comment by nikclayton** [comment, marked as answer]
Dupe of

- https://github.com/orgs/community/discussions/9752
- https://github.com/orgs/community/discussions/62263
- https://github.com/orgs/community/discussions/66111
- https://github.com/orgs/community/discussions/17389
- https://github.com/orgs/community/discussions/27226
